src/lib/RequestParser.ts

```diff
@@ -191,18 +242,29 @@ export class RequestParser {
                     flags: signTransactionRequest.flags,
                     validityStartHeight: signTransactionRequest.validityStartHeight,
                 } as SignTransactionRequest;
-            case RequestType.CASHLINK:
-                const cashlinkRequest = request as ParsedCashlinkRequest;
+            case RequestType.CREATE_CASHLINK:
+                const createCashlinkRequest = request as ParsedCreateCashlinkRequest;
+                // Note that there is no need to export autoTruncateMessage as the message already got truncated
                 return {
-                    appName: cashlinkRequest.appName,
-                    cashlinkAddress: cashlinkRequest.cashlinkAddress
-                            ? cashlinkRequest.cashlinkAddress.toUserFriendlyAddress()
-                            : undefined,
-                    senderAddress: cashlinkRequest.senderAddress
-                            ? cashlinkRequest.senderAddress.toUserFriendlyAddress()
+                    appName: createCashlinkRequest.appName,
+                    senderAddress: createCashlinkRequest.senderAddress
+                            ? createCashlinkRequest.senderAddress.toUserFriendlyAddress()
                             : undefined,
-                    senderBalance: cashlinkRequest.senderBalance,
-                } as CashlinkRequest;
+                    senderBalance: createCashlinkRequest.senderBalance,
+                    value: createCashlinkRequest.value,
+                    message: createCashlinkRequest.message,
+                    theme: createCashlinkRequest.theme,
+                    returnCashlink: createCashlinkRequest.returnCashlink,
+                    skipSharing: createCashlinkRequest.skipSharing,
+                } as CreateCashlinkRequest;
```

Comment:
Can possibly be shortened with object spreading.

Reply:
This also holds true for most other requests.
Can be addressed as part of https://github.com/nimiq/hub/issues/358.